enhancements/multi-arch/heterogeneous-architecture.md

```diff
@@ -0,0 +1,349 @@
+with an ARM worker. As a workaround until imagestreams support importing all the architectures, they can be provided the manifest listed image with the
+`--image` parameter which would allow the commands to work on the worker nodes as well.
+
+### OLM
+
+OLM would need to make sure that it deploys operators only to nodes whose architecture matches the architectures on which the operator is supported.
+This would mean that the OLM would have to be told which architectures an operator supports by the Console operator. Today, OLM has a subscription
+object which is used to tell it how to deploy an operator. It has support for the nodeselector field but this would not be sufficient to specify
+filtering on multiple architectures. To this end, it would have to support the podAffinity field which uses the nodeselectorterms to filter on
+multiple values.
+
+### Machine Config Operator
+
+No changes are needed in the MCO for Phase 1 as the machine-os-content image would be a manifest listed image and the machine config daemon would
+extract the relevant architecture's machine-os-content based on the node it runs on. In the future, architecture specific configurations should
+not be required, instead they need to be templatized and be generalized for certain items, for example kubelet system reserved memory could scale
+based on page size.
+
+### Hypershift
+
+The addition of heterogeneous support in Openshift also opens up a new dimension for [support][OCPPLAN-5684] in Hypershift where the control plane 
+and the workers could run on different architectures. There would be certain changes needed to realize this as there are limitations which need to
+be addressed:
+
+- When creating the hosted cluster, we need to override the `--release-image` option to specify the manifest listed release image in order
+to be able to deploy on a worker node of any architecture. Optionally, the config could provide a flag to indicate that it is being installed
+on a heterogeneous architecture cluster in which case it will pull down a manifest listed release payload rather than the x86 release payload.
+- For the hosted control plane to work on non x86 architecture nodes, the following image references would need changing:
+  - the release image fixtures referencing x86 imagestreams. these would need to be changed to manifestlist references or there would need
+  to be imagestreams added for individual architectures and they would be picked based on the node architecture.
+  - references to some images like cluster-api-provider-aws, apiserver-network-proxy etc.. are x86. these would need to change to provide the images
+  for other architectures too. These images can be overridden through annotations today.
+- Another option would be to mandate an x86 worker node be present for Hypershift and nodeselect the x86 node for the hosted cluster control plane
```

Comment:
>Another option would be to mandate an x86 worker node be present for Hypershift and nodeselect the x86 node for the hosted cluster control plane

I think this is a non-go, A Hosted Cluster might require multiple Nodes e.g HA and the coupling with the arch is unnecessary. We'll need to make sure statically pinned Images are multi-arch

Reply:
will remove this as an option.